### PR TITLE
[2FA][ci] update documentation for CI after latest 2FA changes

### LINKED ISSUE #938: Doc recommends Apple ID without 2FA enabled but doesn't say how
Referring to:
https://docs.fastlane.tools/best-practices/continuous-integration/

"The easiest way to get fastlane running on a CI system is to create a separate Apple ID that doesn't have 2-factor authentication enabled"

It may be the "easiest" way but there's no description of how to do it, and as far as I can tell it's currently impossible?  If there is a way, I haven't discovered it.

### PR DESCRIPTION
**Pull request by rogerluan**
Resolves https://github.com/fastlane/docs/issues/938
Resolves https://github.com/fastlane/docs/issues/959
Resolves https://github.com/fastlane/docs/issues/913
Resolves https://github.com/fastlane/docs/issues/1005

### CODE REVIEW

**Review by ainame — COMMENTED** [review]
I will look into details but regarding four methods, I think we should place the recommended way at first, shouldn't we? 

How about this order?

1. ASC API key (recommended) - you should try this first 
2. Two-step or two factor - you can try this if you are not happy with 1.
3. App specific password - it's for limited usage but still works
4. Apple ID without 2FA (deprecated) - explicitly noted as it's died for people still wondering this solution lol

**Comment by rogerluan** [comment]
@ainame yeah I kinda agree! I debated over the order as well 😂 My initial rationale was: I want to make people fully aware of all the methods before getting to the preferred one (and then leaving the deprecated one last, just for historical purposes). But I guess yours makes sense as people are not interested in history, they want the best solution as fast as possible 😂 

Your order thus makes sense to me! I was also thinking of rephrasing `deprecated` to something like `no longer supported`, what do you think? Is there a technical difference? Cuz it's not actually "deprecated" (as in you can use but we won't maintain it), it's really a "deceased" method 😅

**Comment by ainame** [comment]
@rogerluan yay 🙌 

Regarding the term "deprecated", I can't tell the difference in the native speaker sense 😂  but "obsoleted" would be more appropriate in this context? It was there but no longer in use or work

**Comment by rogerluan** [comment]
@ainame after an internal mental debate, I decided to leave it as `deprecated` anyway 😂

**Review by ainame — APPROVED** [review]
LGTM with a minor suggestion 👍 

Well done.

File: docs/best-practices/continuous-integration.md [hunk 1]
```diff
@@ -21,68 +21,63 @@ Multiple CI products and services offer integrations with fastlane:
...
+## Authenticating with Apple services
 
-Several Fastlane actions communicate with Apple services that need authentication. This can pose several challenges on CI:
+Several _fastlane_ actions communicate with Apple services that need authentication. This can pose several challenges on CI. There are 3 ways to connect to Apple services:
 
-### Separate Apple ID for CI
+### Method 1: App Store Connect API key (recommended)
 
-The easiest way to get _fastlane_ running on a CI system is to create a separate Apple ID that
+This is the recommended and official way of authenticating with Apple services. However, it doesn't support all features yet. Check out [App Store Connect API](https://docs.fastlane.tools/app-store-connect-api/) for more information.
```
[the remark this answers is not in the document]
Comment (rogerluan) [reply docs/best-practices/continuous-integration.md]: Good point! I edited your suggestion to italicize _fastlane_, and applied it 🤗 Thank you!

**Pull request merged**